libs/designsystem/src/lib/components/form-field/directives/date/date-input.directive.ts

```diff
@@ -0,0 +1,97 @@
+import { FormatWidth, getLocaleDateFormat } from '@angular/common';
+import {
+  AfterViewInit,
+  Directive,
+  ElementRef,
+  HostBinding,
+  HostListener,
+  Inject,
+  LOCALE_ID,
+  Renderer2,
+} from '@angular/core';
+import Inputmask from 'inputmask';
+
+@Directive({
+  selector: '[kirby-input][type="date"]',
+})
+export class DateInputDirective implements AfterViewInit {
+  // Ensure numeric keyboard
+  @HostBinding('attr.inputmode') inputmode = 'numeric';
+
+  @HostListener('input')
+  onInput() {
+    this.updateMask(this.elementRef.nativeElement.value);
+  }
+
+  private maskingElement: HTMLElement;
+
+  constructor(
+    private elementRef: ElementRef,
+    private renderer: Renderer2,
+    @Inject(LOCALE_ID) private locale: string
+  ) {
+    // Remove type to avoid user-agent specific behaviour for [type="date"]
+    this.elementRef.nativeElement.removeAttribute('type');
+  }
+
+  ngAfterViewInit(): void {
+    this.initMask();
+  }
+
+  private initMask(): void {
+    const inputFormat = this.getInputFormat();
+    const placeholder = this.getPlaceholder(inputFormat);
+
+    // Set initial placeholder ex. dd/mm/yyyy
+    this.renderer.setAttribute(this.elementRef.nativeElement, 'placeholder', placeholder);
+
+    // Init InputMask
+    new Inputmask('datetime', {
+      inputFormat,
+      placeholder,
+    }).mask(this.elementRef.nativeElement);
+
+    // Append input overlay, so it's possible to style typed date differntly than the date-mask
+    this.appendMaskingElement();
+  }
+
+  // Keeps order and seperator from speficied locale
+  private getInputFormat(): string {
+    const localeDateFormat = getLocaleDateFormat(this.locale, FormatWidth.Short);
+    return localeDateFormat
+      .toLowerCase()
+      .replace(/d+/, 'dd')
+      .replace(/m+/, 'mm')
+      .replace(/y+/, 'yyyy');
+  }
+
+  private getPlaceholder(inputFormat: string): string {
+    return this.locale === 'da' ? inputFormat.split('y').join('å') : inputFormat;
```

Comment:
Did you create a separate issue for handling locales?
This construct can get ugly real fast if/when we need to support more languages. I think it would be better to create a central language service/helper - even if it is very simple at this point. Would also make it easier to unit test these transformations in isolation

Reply:
Totally agree, and yes I did it's right here #1450